### ISSUE TITLE
Primary tablet should not take write when mysqld is down or replication is not setup

### ISSUE THREAD

**Issue by narcsfz** [vitessio/vitess#8132]
#### Feature Description
Consider the following events:
T1: mysqld on the primary tablet (A) crashed, replications stop
T2: mysqld restarted
T3: an write request arrives
T4: the old primary (A) crashed and we execute a reparent command to switch primary
T5: replicas setup connection with the new primary

Vttablet has a polling mechanism to detect mysqld is down, but I'm not 100% sure if it will immediately broadcast this information to vtgate and set is_serving to false (appreciate a code pointer if we already do so)

However, until the replication is setup, those writes can potentially lost, I think VTTablet should consider a mysqld without replication as if it is down and fail the request directly

#### Use Case(s)
See description above

**Comment by aquarapid** [comment]
Couple of observations here:
  * This is a case that the Vitess semi-sync support will naturally avoid, since the primary will not accept writes until a replica is connected (and ack-ing writes)
  * Even without semi-sync;  this should not happen normally, since you would (should?) configure your MySQL instances to start up as read-only;  so after a crash and restart, they cannot accept writes until they are promoted back to primary.

**Comment by sonne5** [comment]
> Even without semi-sync; this should not happen normally, since you would (should?) configure your MySQL instances to start up as read-only; so after a crash and restart, they cannot accept writes until they are promoted back to primary.

Yep, that is correct. Setting mysqld to start with read-only can address the issue. I'm wondering if it would be helpful optimization to make vttable to be more defensive in this case if mysqld is _not_ configured to start with read_only.

**Comment by aquarapid** [comment]
Yeah, I do think that adding that check (probably as an optional parameter), could be useful for additional safety for users not using `mysqlctl`.  Most users are already getting the effective benefit of this if they are running `mysqlctl` to manage their MySQL instance, since the default `my.cnf` template config (`config/mycnf/default.cnf`), already specifies the server to start with server option `read-only`.

**Comment by sonne5** [comment]
@aquarapid one quick follow up: if the topo server has staled record saying a node is primary, and will Vitess set the mysqld with read_only=false after it restarts?

**Comment by aquarapid** [comment]
No;  in general if `mysqld` restarts underneath a vttablet, and is using the default `my.cnf` template that includes the `read-only` directive;  it will remain in `read-only`, even if it is a (valid or invalid) primary, until you actively perform a PRS or something similar (e.g. via orchestrator).
